source/features/clone-branch.tsx

```diff
@@ -0,0 +1,84 @@
+import gitBranch from 'octicon/git-branch.svg';
+import React from 'dom-chef';
+import select from 'select-dom';
+import delegate, {DelegateEvent} from 'delegate-it';
+import * as api from '../libs/api';
+import features from '../libs/features';
+import {getRepoURL} from '../libs/utils';
+
+async function cloneBranch(event: DelegateEvent<MouseEvent, HTMLAnchorElement>): Promise<void | false> {
+	const currentTarget = event.delegateTarget;
+	const branchElement = currentTarget.closest<HTMLAnchorElement>('[data-branch-name]');
+
+	// eslint-disable-next-line no-control-regex
+	const invalidBranchName = new RegExp(/^[./]|\.\.|@{|[/.]$|^@$|[~^:\u0000-\u0020\u007F\s?*]/); // Taken from https://stackoverflow.com/questions/3651860/which-characters-are-illegal-within-a-branch-name#comment71900602_3651867
+	let newBranchName = prompt('Enter the new branch name')?.trim();
+
+	while (invalidBranchName.test(newBranchName as string)) {
+		newBranchName = prompt('Unsupported branch name (https://git-scm.com/docs/git-check-ref-format) \nEnter the new branch name', newBranchName)?.trim();
+	}
+
+	if (!newBranchName) {
+		return false;
+	}
+
+	const spinner = select('.js-loading-spinner', branchElement!)!;
+	spinner.hidden = false;
+	currentTarget.hidden = true;
+
+	try {
+		const getBranchInfo = await api.v3(`repos/${getRepoURL()}/git/refs/heads/${branchElement!.dataset.branchName!}`);
+		await api.v3(`repos/${getRepoURL()}/git/refs`, {
+			method: 'POST',
+			body: {
+				sha: getBranchInfo.object.sha,
+				ref: 'refs/heads/' + newBranchName
+			}
+		});
+		const redirectURL = new URL(location.href);
+		redirectURL.pathname = redirectURL.pathname.endsWith('/all') ? redirectURL.pathname : redirectURL.pathname + '/all';
+		const search = new URLSearchParams(location.search);
+		search.set('query', newBranchName);
+		redirectURL.search = String(search);
+		location.href = String(redirectURL);
```

Comment:
I am not feeling well my mind is not working to write this better

Reply:
Verbose but it looks fine. Only this is necessary
```suggestion
		location.replace(String(redirectURL));
```